docs/InMemoryTokenSelection.md

```diff
@@ -119,3 +137,17 @@ addTokensToRedeem(
     changeOutput = changeOutput
 )
 ```
+
+
+### Providing Queries to LocalTokenSelector
+
+Query utility functions in `om.r3.corda.lib.tokens.selection` (`tokenAmountCriteria`, `tokenAmountWithIssuerCriteria`, `tokenAmountWithHolderCriteria`) can be useful in building `TokenQueryBy` 
+instances for  `LocalTokenSelector`, for example:
+
+```kotlin
+// Get list of input and output states that can be passed to addMove or MoveTokensFlow
+val (inputs, outputs) = localTokenSelector.generateMove(
+        partiesAndAmounts = listOf(Pair(receivingParty,  tokensAmount)),
+        changeHolder = this.ourIdentity,
+        queryBy = TokenQueryBy(queryCriteria = tokenAmountCriteria(tokensAmount.token)))
```

Comment:
This is almost there. I forgot to add that to the docs! The only issue is that `queryCriteria` was left because we exposed it in flows API. Local token selection doesn't use this field at all. Only `issue`r and `predicate`. This will be changed in 2.0 (with API break) for now it's ugly.

Reply:
@kasiastreich I couldn't make `MoveTokensFlow` work without adding the `TokenQueryBy`. We issue fungible tokens based on an `EvolvableTokenType`, so trying to move without the `queryBy` would give an error:

```
com.r3.corda.lib.tokens.selection.InsufficientBalanceException: Insufficient spendable states identified for 1.00 TokenPointer(class my.EvolvableTokenType, daec0035-b2e6-42ef-8dbd-c454e877aec2).
	at com.r3.corda.lib.tokens.selection.memory.services.VaultWatcherService.selectTokens(VaultWatcherService.kt:168) ~[?:?]
	at com.r3.corda.lib.tokens.selection.memory.selector.LocalTokenSelector.selectTokens(LocalTokenSelector.kt:56) ~[?:?]
	at com.r3.corda.lib.tokens.selection.api.Selector.generateMove(Selector.kt:177) ~[?:?]
	at com.r3.corda.lib.tokens.selection.api.Selector.generateMove(Selector.kt:112) ~[?:?]
	at com.r3.corda.lib.tokens.selection.api.Selector.generateMove$default(Selector.kt:110) ~[?:?]
	at io.ivno.tokenpoc.cordapp.workflow.flow.transfer.MoveFungibleTokenSamplerFlow.call(MoveFungibleTokenFlow.kt:101) ~[?:?]
	at io.ivno.tokenpoc.cordapp.workflow.flow.transfer.MoveFungibleTokenSamplerFlow.call(MoveFungibleTokenFlow.kt:68) ~[?:?]
	at net.corda.node.services.statemachine.FlowStateMachineImpl.run(FlowStateMachineImpl.kt:273) ~[corda-node-4.2.jar:?]
	at net.corda.node.services.statemachine.FlowStateMachineImpl.run(FlowStateMachineImpl.kt:45) ~[corda-node-4.2.jar:?]
	at co.paralleluniverse.fibers.Fiber.run1(Fiber.java:1092) ~[quasar-core-0.7.11_r3-jdk8.jar:0.7.11_r3]
	at co.paralleluniverse.fibers.Fiber.exec(Fiber.java:788) ~[quasar-core-0.7.11_r3-jdk8.jar:0.7.11_r3]
	at co.paralleluniverse.fibers.RunnableFiberTask.doExec(RunnableFiberTask.java:100) ~[quasar-core-0.7.11_r3-jdk8.jar:0.7.11_r3]
	at co.paralleluniverse.fibers.RunnableFiberTask.run(RunnableFiberTask.java:91) ~[quasar-core-0.7.11_r3-jdk8.jar:0.7.11_r3]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_212]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.29.Final.jar:4.1.29.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

Reply:
Ok, thanks, I will take a look, it can be a bug.

Reply:
Hey :) I just had a moment to come back to this PR. So, from your comments, are you using MoveTokensFlow? Or are you using addMoveX manually? Because MoveTokensFlow will always use database selection. If you would like to use local token selector you need to assemble your move flows yourself.

Reply:
Hey @kasiastreich , current;y using `addMoveFungibleTokens`. We were doing some performance testing and database selection was complaining about insufficient funds, but thankfully this was about the time in-memory selection was made available ;-)

Reply:
ok, I know what happened :) `addMoveFungibleTokens` uses database - that's why you had to provide query criteria. I can see now it is confusing. To use in memory token selection it is sufficient to construct `LocalTokenSelector`, use `select` or `generateMove` methods, then add the states to the transaction builder using `addMoveTokens`. I will try to phrase it better in the documentation!